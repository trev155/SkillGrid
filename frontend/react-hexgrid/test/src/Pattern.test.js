import React from 'react';
import renderer from 'react-test-renderer';

import Pattern from '../../src/Pattern';

test('Pattern should render correctly with default props', () => {
  const tree = renderer.create(
    <Pattern
      id={'test1'}
      link={'http://example.com'}
      size={{ x: 100, y: 200 }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Pattern should render correctly with custom props', () => {
  const tree = renderer.create(
    <Pattern
      id={'test2'}
      link={'http://example.com'}
      size={{ x: 200, y: 100 }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
