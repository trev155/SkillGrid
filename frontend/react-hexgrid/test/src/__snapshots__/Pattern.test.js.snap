// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pattern should render correctly with custom props 1`] = `
<defs>
  <pattern
    height={100}
    id="test2"
    patternUnits="objectBoundingBox"
    width={200}
    x={0}
    y={0}
  >
    <image
      height={200}
      width={400}
      x={0}
      xlinkHref="http://example.com"
      y={0}
    />
  </pattern>
</defs>
`;

exports[`Pattern should render correctly with default props 1`] = `
<defs>
  <pattern
    height={200}
    id="test1"
    patternUnits="objectBoundingBox"
    width={100}
    x={0}
    y={0}
  >
    <image
      height={400}
      width={200}
      x={0}
      xlinkHref="http://example.com"
      y={0}
    />
  </pattern>
</defs>
`;
